packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2004,6 +2062,135 @@ export class TransactionController extends BaseController<
   private getGasFeeFlows(): GasFeeFlow[] {
     return [new LineaGasFeeFlow()];
   }
+
+  async #updateSimulationData(transactionMeta: TransactionMeta) {
+    const { id: transactionId, chainId, txParams } = transactionMeta;
+    const { from, to, value, data } = txParams;
+
+    let simulationData: SimulationData = {
+      error: {
+        code: SimulationErrorCode.Disabled,
+        message: 'Simulation disabled',
+      },
+      tokenBalanceChanges: [],
+    };
+
+    if (this.#isSimulationEnabled()) {
+      this.#updateTransactionInternal(
+        { transactionId, skipHistory: true },
+        (txMeta) => {
+          txMeta.simulationData = undefined;
+        },
+      );
+
+      simulationData = await getSimulationData({
+        chainId,
+        from: from as Hex,
+        to: to as Hex,
+        value: value as Hex,
+        data: data as Hex,
+      });
+    }
+
+    const finalTransactionMeta = this.getTransaction(transactionId);
+
+    if (!finalTransactionMeta) {
+      log(
+        'Cannot update simulation data as transaction not found',
+        transactionId,
+        simulationData,
+      );
+
+      return;
+    }
+
+    this.#updateTransactionInternal(
+      {
+        transactionId,
+        note: 'TransactionController#updateSimulationData - Update simulation data',
+      },
+      (txMeta) => {
+        txMeta.simulationData = simulationData;
+      },
+    );
+
+    log('Updated simulation data', transactionId, simulationData);
+  }
+
+  #updateTransactionInternal(
+    {
+      transactionId,
+      note,
+      skipHistory,
+    }: { transactionId: string; note?: string; skipHistory?: boolean },
+    callback: (transactionMeta: TransactionMeta) => TransactionMeta | void,
+  ) {
+    let updatedTransactionParams: (keyof TransactionParams)[] = [];
+
+    this.update((state) => {
+      const index = state.transactions.findIndex(
+        ({ id }) => id === transactionId,
+      );
+
+      let transactionMeta = state.transactions[index];
+
+      // eslint-disable-next-line n/callback-return
+      transactionMeta = callback(transactionMeta) ?? transactionMeta;
+
+      transactionMeta.txParams = normalizeTransactionParams(
+        transactionMeta.txParams,
+      );
+
+      validateTxParams(transactionMeta.txParams);
+
+      updatedTransactionParams =
+        this.#checkIfTransactionParamsUpdated(transactionMeta);
+
+      if (skipHistory !== true) {
+        transactionMeta = updateTransactionHistory(
+          transactionMeta,
+          note ?? 'Transaction updated',
+        );
+      }
+
+      state.transactions[index] = transactionMeta;
+    });
+
+    const transactionMeta = this.getTransaction(
+      transactionId,
+    ) as TransactionMeta;
+
+    if (updatedTransactionParams.length > 0) {
+      this.#onTransactionParamsUpdated(
+        transactionMeta,
+        updatedTransactionParams,
+      );
+    }
+  }
+
+  #onGasFeePollerTransactionUpdate({
+    transactionId,
```

Comment:
If this was previously a call to `updateTransaction` and therefore a `TransactionMetadata` object, then won't the ID property be `id` rather than `transactionId`?

Reply:
True, I didn't notice that, thanks